ideascube/forms.py

```diff
@@ -70,14 +70,55 @@ def save(self, *args, **kwargs):
 class UserImportForm(forms.Form):
     source = forms.FileField(required=True)
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        formats = getattr(settings, 'USER_IMPORT_FORMATS', None)
+
+        if formats is None:
+            self.fields['format'] = forms.CharField(
+                required=True, initial='ideascube', widget=forms.HiddenInput)
+
+        else:
+            self.fields['format'] = forms.ChoiceField(
+                required=True, initial=formats[0][0], choices=formats)
```

Comment:
So first, it's a case which would only happen if the box is widely misconfigured. I added it for safety, but it really won't ever happen (I know, "famous last words" 😛)

And indeed, you're right that the `ValidationError` as I did it is quite user-hostile.

However, the `ValidationError` tells the user that something went wrong, and it might push them to come and tell us that something is bad with the box, which would lead us to find (and fix) the misconfiguration. Simply hiding the option as you suggest would silently ignore the misconfiguration issue, users might try to import anyway with the wrong mapping, and things might get ugly real quick.

I just thought about something: how about I add a test to `ideascube/tests/test_settings.py` which ensures that all the declared `USER_IMPORT_FORMATS` have a corresponding handler in the code?

This way, we can just remove the `ValidationError` code path, we just know that it won't happen.